modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicReader.java

```diff
@@ -549,47 +550,71 @@ private void initReaderFromURL(final Object source, final Hints hints) throws Ex
-                //
-                final File parent = URLs.urlToFile(sourceURL).getParentFile();
-
-                // this can be used to look for properties files that do NOT define a datastore
-                final File[] properties =
-                        parent.listFiles(
-                                (FilenameFilter)
-                                        FileFilterUtils.and(
-                                                FileFilterUtils.notFileFilter(
-                                                        FileFilterUtils.nameFileFilter(
-                                                                "indexer.properties")),
-                                                FileFilterUtils.and(
-                                                        FileFilterUtils.notFileFilter(
-                                                                FileFilterUtils.nameFileFilter(
-                                                                        "datastore.properties")),
-                                                        FileFilterUtils.makeFileOnly(
-                                                                FileFilterUtils.suffixFileFilter(
-                                                                        ".properties")))));
-
-                // do we have a valid datastore + mosaic properties pair?
-                final File datastoreProperties = new File(parent, "datastore.properties");
-
-                // Scan for MosaicConfigurationBeans from properties files
+            if (configuration != null) {
+                // Old style code: we have a single MosaicConfigurationBean. Use that
+                // to create the catalog
+                granuleCatalog =
+                        ImageMosaicConfigHandler.createCatalog(
+                                sourceURL, configuration, this.hints);
+                File parent = URLs.urlToFile(sourceURL).getParentFile();
+                MultiLevelROIProvider rois =
+                        MultiLevelROIProviderMosaicFactory.createFootprintProvider(parent);
+                granuleCatalog.setMultiScaleROIProvider(rois);
+                addRasterManager(configuration, true);
+            } else {
+                // 2nd attempt: look for a property file with same name of the mosaic
                 List<MosaicConfigurationBean> beans = new ArrayList<>();
-                if (properties != null) {
-                    for (File propFile : properties) {
-                        if (Utils.checkFileReadable(propFile)
-                                && Utils.loadMosaicProperties(URLs.fileToUrl(propFile)) != null) {
-                            configuration = Utils.loadMosaicProperties(URLs.fileToUrl(propFile));
-                            if (configuration != null) {
-                                beans.add(configuration);
+                if (configuration == null
```

Comment:
So, if I understand correctly, it first attempt to locate the configuration file using the parent folder name, and stops there if found (common case). If not, goes for the uncommon case, looking for a valid config file in property form with any other name?

Reply:
yep.
So if you have
myMosaic/myMosaic.properties
myMosaic/file1.tif
myMosaic/file2.tif
...
myMosaic/file20000.tif
It will quickly look for a myMosaic/myMosaic.properties file.

if missing, it will go through current behaviour via a listFiles.(*.properties as filefilter)